source/source.go

```diff
@@ -20,71 +20,53 @@ import (
 	"fmt"
 
 	"cloud.google.com/go/storage"
-	"github.com/conduitio-labs/conduit-connector-google-cloudstorage/config"
-	srcConfig "github.com/conduitio-labs/conduit-connector-google-cloudstorage/source/config"
+	"github.com/conduitio-labs/conduit-connector-google-cloudstorage/source/config"
 	"github.com/conduitio-labs/conduit-connector-google-cloudstorage/source/iterator"
 	"github.com/conduitio-labs/conduit-connector-google-cloudstorage/source/position"
+	commonsConfig "github.com/conduitio/conduit-commons/config"
+	"github.com/conduitio/conduit-commons/opencdc"
 	sdk "github.com/conduitio/conduit-connector-sdk"
 	"google.golang.org/api/option"
 )
 
 // Source connector.
 type Source struct {
 	sdk.UnimplementedSource
-	config   srcConfig.SourceConfig
+	config   config.Config
 	client   *storage.Client
 	iterator Iterator
 }
 
 type Iterator interface {
-	Next(ctx context.Context) (sdk.Record, error)
+	Next(ctx context.Context) (opencdc.Record, error)
 	Stop(ctx context.Context)
 }
 
 func NewSource() sdk.Source {
 	return sdk.SourceWithMiddleware(&Source{}, sdk.DefaultSourceMiddleware()...)
 }
 
-func (s *Source) Parameters() map[string]sdk.Parameter {
-	return map[string]sdk.Parameter{
-		config.ConfigKeyGCPServiceAccountKey: {
-			Default:     "",
-			Required:    true,
-			Description: "Google Cloud Storage ServiceAccountKey",
-		},
-		config.ConfigKeyGCSBucket: {
-			Default:     "",
-			Required:    true,
-			Description: "Google Cloud Storage Bucket",
-		},
-		srcConfig.ConfigKeyPollingPeriod: {
-			Default:     srcConfig.DefaultPollingPeriod,
-			Required:    false,
-			Description: "polling period for the CDC mode, formatted as a time.Duration string.",
-		},
-	}
+func (s *Source) Parameters() commonsConfig.Parameters {
+	return s.config.Parameters()
 }
 
 // Configure parses and stores the configurations
 // returns an error in case of invalid config.
-func (s *Source) Configure(ctx context.Context, cfg map[string]string) error {
+func (s *Source) Configure(ctx context.Context, cfgRaw commonsConfig.Config) error {
 	logger := sdk.Logger(ctx).With().Str("Class", "Source").Str("Method", "Configure").Logger()
 	logger.Trace().Msg("Starting Configuring the Source Connector...")
 
-	sourceConfig, err := srcConfig.ParseSourceConfig(ctx, cfg)
+	err := sdk.Util.ParseConfig(ctx, cfgRaw, &s.config, NewSource().Parameters())
 	if err != nil {
-		logger.Error().Stack().Err(err).Msg("Error While parsing the Source Config")
-		return err
+		return err //nolint: wrapcheck // not needed here
```

Comment:
no harm in wrapping it here, imo it's better to, just a simple `fmt.Errorf("invalid config: %w", err)`